src/components/Home/Highlights/HighlightsManager.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { FC, useEffect } from "react"
+import { View } from "react-native"
+import { PoliCarousel, WidgetType } from "./PoliCarousel"
+import lectures from "./lectures.json"
+import { api, Lecture } from "api"
+
+//ideally this is the object we want to pass to the carousel
+//this might be expansed in the future
+export interface CarouselItem {
+    /**
+     * number to identify the object and the relative widget
+     */
+    id: number
+    /**
+     * enum field to identify the type of the widget
+     */
+    type: WidgetType
+    /**
+     * string to identify the date of the event(or deadline) contained in the widget
+     */
+    date: string
+    /**
+     * string to identify the time of the event(or deadline) contained in the widget
+     */
+    time: string
+    /**
+     * string to identify the title of the event contained in the widget
+     */
+    title: string
+    /**
+     * string to identify eventually the room when the event takes place
+     */
+    room?: string
+}
+
+//just to test with same static data
+const data: CarouselItem[] = [
+    {
+        id: 0,
+        type: WidgetType.ISEE,
+        date: "Mer 22/10/2022",
+        time: "8:00",
+        title: "Scadenza ISEE 2023",
+    },
+    {
+        id: 1,
+        type: WidgetType.ISEE,
+        date: "Mer 22/10/2022",
+        time: "8:00",
+        title: "Scadenza ISEE 2023",
+    },
+    {
+        id: 2,
+        type: WidgetType.ISEE,
+        date: "Mer 22/10/2022",
+        time: "8:00",
+        title: "Scadenza ISEE 2023",
+    },
+]
```

Comment:
`data` deve 100% essere un elemento dello state, ogni elemento al cui variare deve variare anche la UI di conseguenza deve essere nello state, dentro `HighlightsManager` metti:

```ts
const [data, setData] = useState<CarouselItem[]>([
{
        id: 0,
        type: WidgetType.ISEE,
        date: "Mer 22/10/2022",
        time: "8:00",
        title: "Scadenza ISEE 2023",
    },
    {
        id: 1,
        type: WidgetType.ISEE,
        date: "Mer 22/10/2022",
        time: "8:00",
        title: "Scadenza ISEE 2023",
    },
    {
        id: 2,
        type: WidgetType.ISEE,
        date: "Mer 22/10/2022",
        time: "8:00",
        title: "Scadenza ISEE 2023",
    },
])
```

dovrai spostare la definizione della funzione `findNextLecture` all'interno del corpo di `HighlightsManager`, e al posto di
```ts
addItem(nextLecture)
```
dovrai fare
```ts
setData([ nextLecture, ...data ])
```

è molto importante che in React tutti i dati rimangano reattivi, ogni modifica ai dati cambia lo state, e una modifica allo state aggiorna la UI

Reply:
okay grazie mille del consiglio!Ma è normale che se runno l'app e poi faccio Ctrl+S continuano ad aggiungersi widget?